src/main/java/gift/kakaoapi/dto/UserInfo.java

```diff
@@ -0,0 +1,36 @@
+package gift.kakaoapi.dto;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class UserInfo {
+
+    @JsonProperty("id")
+    String id;
+
+    @JsonProperty("kakao_account")
+    public KakaoAccount kakaoAccount;
+
+    @JsonIgnoreProperties(ignoreUnknown = true)
+    public class KakaoAccount {
```

Comment:
```suggestion
    public static class KakaoAccount {}
```
static으로 선언한 것과 아닌 것의 차이는 무엇인가요?

Reply:
Inner Class를 static으로 선언해야하는 이유에 대해서 찾아보았습니다. 

Inner Class를 static으로 선언하지 않으면 외부 클래스와 내부 클래스 사이에서는 "외부 참조"가 발생하게 됩니다. 따라서 내부 클래스의 데이터가 남아있다면(가비지 컬렉션의 대상이 아니라면) 외부 클래스도 남아있기때문에(내부 클래스가 외부 클래스를 참조하고 있음) 메모리 누수가 발생할 수 있습니다. 외부 클래스의 Static 키워드를 활용해 내부 클래스를 정의하게 되면 이런 경우를 대비할 수 있습니다. 또한, 내부 클래스를 static으로 선언 한다면 외부클래스를 거치지 않고 내부 클래스 객체를 독립적으로 생성할 수 있습니다. 

(단, 내부 클래스에서 외부 클래스의 필드에 접근이 필요한 경우에는 static을 붙이지 않고 내부 클래스를 선언해야합니다.)